### PR TITLE
Fix/qr scanner

### PR DESCRIPTION
**Pull request by maycon-mello**

### CODE REVIEW

File: src/features/qr-code-scanner/qr-code.js [hunk 1]
```diff
@@ -20,8 +20,17 @@ export async function credentialHandler(data) {
   try {
     // @ts-ignore
     const credentials: Credentials = Credentials.getInstance();
-    const credentialData = credentials.getCredentialFromUrl(data);
-    await credentials.add(credentialData);
+
```
Comment (mike-parkhill) [inline src/features/qr-code-scanner/qr-code.js:23]: could probably add a null/string check up here and skip the rest if it doesn't pass that simple check.

**Comment by maycon-mello** [comment]
This has https://github.com/docknetwork/dock-app/pull/141 as a dependency

File: src/features/qr-code-scanner/qr-code.js (hunk 1 above)
Comment (maycon-mello) [reply src/features/qr-code-scanner/qr-code.js:49]: Not sure on this, are you referring to the `Credentials.getInstance()`?

Comment (maycon-mello) [reply src/features/qr-code-scanner/qr-code.js:49]: Ah got it, it's the handler. I've pushed an updated for that

**Comment by maycon-mello** [comment]
Added a new commit @mike-parkhill, with better error handlers and tests

File: src/features/credentials/credentials.js [hunk 2]
```diff
@@ -18,6 +18,7 @@ export function useCredentials() {
     await syncCredentials();
   };
 
+  console.log('credetials', items);
```
Comment (mike-parkhill) [inline src/features/credentials/credentials.js:21]: probably meant to delete this?

File: src/features/qr-code-scanner/qr-code.js [hunk 3]
```diff
@@ -34,16 +63,40 @@ export async function credentialHandler(data) {
...
-      return;
+      return true;
     }
   }
+
+  return false;
+}
+
+export async function qrCodeHandler(data, handlers = qrCodeHandlers) {
+  const success = await executeHandlers(data, handlers);
```
Comment (mike-parkhill) [inline src/features/qr-code-scanner/qr-code.js:94]: validate `data` is defined?

Comment (maycon-mello) [reply src/features/qr-code-scanner/qr-code.js:94]: its validated inside of executeHandlers already

File: src/features/qr-code-scanner/qr-code.test.js [hunk 4]
```diff
@@ -30,6 +42,27 @@ describe('qr-code', () => {
...
+
+    it('expect not to trigger handlers for empty data', async () => {
+      const data = '';
+      const mockHandlers = [jest.fn()];
+      await qrCodeHandler(data, mockHandlers);
+
+      expect(mockHandlers[0]).not.toBeCalled();
+    });
+
+    it('expect not to show error toast for not handled data', async () => {
```
Comment (mike-parkhill) [inline src/features/qr-code-scanner/qr-code.test.js:54]: do we give the user any indication that something went wrong?

**Review by mike-parkhill — APPROVED** [review]

**Pull request merged**